### DISCUSSION TITLE
MINOR: gridX and gridY returned as strings by application/geo+json GETs to /gridpoints/{wfo}/{x},{y}

### DISCUSSION THREAD

**Discussion by TimJohns** [weather-gov/api#768] [General]
OK, FIRST of all, thank your for this rather awesome API to this incredible data source! 

While experimenting with the API, we noticed what may be a (very) minor issue with the typing of a couple of returned values. Specifically, the gridX and gridY values returned by GETs to the /gridpoints/{wfo}/{x},{y} endpoint appear to be returned as strings, but they appear to be specified in the OpenAPI doc Schema and example as integers. (Note: The /points endpoint returns them as numbers; we didn't check other endpoints). This is certainly not impacting operationally or blocking any progress, and was only a (very) minor brief distraction while experimenting with the API. 
 
### To Repro
`curl -X GET "https://api.weather.gov/gridpoints/SEW/127,78" -H "accept: application/geo+json"`

---
### Expected
```JSON
...
 "properties": {
    "@id": "https://api.weather.gov/gridpoints/SEW/127,78",
...
    "gridId": "SEW",
    "gridX": 127,
    "gridY": 78,
...
```
---
### Actual
```JSON
...
 "properties": {
    "@id": "https://api.weather.gov/gridpoints/SEW/127,78",
...
    "gridId": "SEW",
    "gridX": "127",
    "gridY": "78",
...
```

**Background:** I'm the founder of Hillwood Park, makers of Epic Road Trip Planner. In the wake of Hurricane Helene, and with Hurricane Milton approaching the Florida cost, we manually added a set of warnings to Epic Road Trip Planner for folks potentially traveling in that region. Manually adding warnings like this doesn't scale, and this morning someone asked what it would take to integrate NWS hazard warnings (and other weather data), and we began experimenting with this API. Short answer is that while there's a lot of work to do on our end before we have that actually integrated, it's clear the API will support what we're trying to do, and I'm super-grateful for that. Folks planning road trips will be safer, more comfortable, and better informed as a result.

**Comment by sullynole** [comment]
Thank you, this discrepancy  has been documented for correction.

**Comment by JoshStern** [comment]
Adding a vote for this as well! @sullynole is there a timeline for this? In the meantime it may be helpful to update the openapi definition to protect clients from the discrepancy (I can open a PR if that would be helpful). It may take some migration to avoid breaking clients that already expect a string.

Also second the sentiment that this is an amazing service to provide and I am really grateful for the work you all are doing.

**Reply by JoshStern** [reply]
I've created this PR for the openapi definition change: https://github.com/weather-gov/api/pull/798

Bundled it with another issue I ran into but please let me know if you'd prefer them to be split.

**Reply by scadergit** [reply]
Note that the code in this repo is for github presentation, and does not include the application itself. The pull request is helpful for discussion, but even if accepted it would not merge directly into the actual code base.

**Reply by JoshStern** [reply]
Not a problem, what is the process for adding it into the distributed openapi definition? If there's another repo I can open a PR there.

**Reply by scadergit** [reply]
Seeing @sullynole's update below made me realize I never answered this question, sorry. The app is an internal project at this time.

**Comment by sullynole** [comment]
This will be resolved in the next release, which is targeted for the week of 5/19 if approved.
